Core/include/Acts/TrackFinding/TrackSelector.hpp

```diff
@@ -340,38 +346,58 @@ bool TrackSelector::isValidTrack(const track_proxy_t& track) const {
   };
 
   const auto theta = track.theta();
-  const auto eta = -std::log(std::tan(theta / 2));
-  const auto absEta = std::abs(eta);
 
-  if (absEta < m_cfg.absEtaEdges.front() ||
-      absEta >= m_cfg.absEtaEdges.back()) {
-    return false;
+  constexpr double kUnset = -std::numeric_limits<double>::infinity();
+
+  double _eta = kUnset;
+  double _absEta = kUnset;
+
+  auto absEta = [&]() {
+    if (_absEta == kUnset) {
+      _eta = -std::log(std::tan(theta / 2));
+      _absEta = std::abs(_eta);
+    }
+    return _absEta;
+  };
+
+  const Config* cutsPtr{nullptr};
+  if (!m_isUnbinned) {
+    if (absEta() < m_cfg.absEtaEdges.front() ||
+        _absEta >= m_cfg.absEtaEdges.back()) {
+      return false;
+    }
+    cutsPtr = &m_cfg.getCuts(_eta);
+  } else {
+    cutsPtr = &m_cfg.cutSets.front();
   }
 
-  const CutConfig& cuts = m_cfg.getCuts(eta);
+  const Config& cuts = *cutsPtr;
 
   return within(track.transverseMomentum(), cuts.ptMin, cuts.ptMax) and
-         within(std::abs(eta), cuts.absEtaMin, cuts.absEtaMax) and
-         within(eta, cuts.etaMin, cuts.etaMax) and
+         within(absEta(), cuts.absEtaMin, cuts.absEtaMax) and
+         within(_eta, cuts.etaMin, cuts.etaMax) and
```

Comment:
Doesn't this still calculate `_eta` when there is no eta cut? How about:
```suggestion
         (noEtaCuts ||
         within(absEta(), cuts.absEtaMin, cuts.absEtaMax) and
         within(_eta, cuts.etaMin, cuts.etaMax)) and
```
where `noEtaCuts` is fairly simply (but verbosely) calculated. This could also be stored in the `Config` for speed, or combined with `m_isUnbinned` by making that an `enum`.

It will still skip `absEta()` if it the pT cut fails. I guess you could quickly gain a bit more by moving this to the last check.

Reply:
Okay, I pushed another version where the `TrackSelector` constructor calculates `m_noEtaCuts` that is then used in the `isValidTrack` function.